CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ option(LUAU_BUILD_WEB "Build Web module" OFF)
 option(LUAU_WERROR "Warnings as errors" OFF)
 option(LUAU_STATIC_CRT "Link with the static CRT (/MT)" OFF)
 option(LUAU_EXTERN_C "Use extern C for all APIs" OFF)
+option(LUAU_EXPORT_DLL "Exports to DLL" OFF)
```

Comment:
Might be better to keep close to CMake and call it LUAU_BUILD_SHARED.

Would also be nice to detect a configuration failure when `LUAU_EXTERN_C` is OFF and `LUAU_BUILD_SHARED` is ON, since that will result in a set of shared libraries that are not very useful.